sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/ConcurrencyControl.java

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.models;
+
+import com.azure.storage.blob.BlobClient;
+
+/**
+ * Defines values for ConcurrencyControl.
+ * {@link #NONE}
+ * {@link #E_TAG}
+ * {@link #VERSION_ID}
+ */
+public enum ConcurrencyControl {
+    /**
+     * No concurrency control. 'requestConditions.ifMatch' and 'client.versionId' must not be set.
+     */
+    NONE,
+
+    /**
+     * Default value. Concurrency control based on eTag.
+     * If {@link BlobRequestConditions#getIfMatch()} is set, the client will honor this value.
+     * Otherwise, {@link BlobRequestConditions#getIfMatch()} is set to the latest eTag.
+     */
+    E_TAG,
```

Comment:
`E_TAG` or `ETAG`? In other locations in Storage we already use constants where it doesn't have the `_` such as https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/Constants.java#L221

Reply:
I think the docs should also specify that we will respect the conditions they pass initially and then we will lock on the etag

Reply:
And should maybe mention that specifying an etag is mutually exclusive to having a version set.

Reply:
I can change it to ETAG

Reply:
@rickle-msft - Does this line not indicate that we will respect their conditions?
If {@link BlobRequestConditions#getIfMatch()} is set, the client will honor this value.
   Otherwise, {@link BlobRequestConditions#getIfMatch()} is set to the latest eTag

Reply:
Yea that's fine. I think when I wrote that, I was thinking about the other comment I left about using the customer's request for the getProperties and applying that here. But anyway. Yea this is fine.

Reply:
Resolved